### ISSUE TITLE
Refactorization of test system

### ISSUE THREAD

**Issue by mdzik** [CFD-GO/TCLB#269]
We the people:
1. need one repo - subrepo makes a lot of headaches and leads to inconsistent structure
2. wherever possible packages installed during the test should have version specified - reproducibility

therefore we the people demand that TCLB_test would be merged into main repo ;)

**Comment by llaniewski** [comment]
@mdzik I agree that git submodules are a pain to use. But I think including test results in main repo would be heavy on all of us. Especially as I would want to completely move to value-based test completely.

I propose we get rid of the tests submodule all together, and add our own way of keeping track of which tests are up to date. And let the test scripts take care of checking out the right repo, commit etc.

**Comment by mdzik** [comment]
We could add some sort of config file where we would set something like 
```
[test1.xml]
ref_data_url="gist/tclb/test_scmp_v11.c.tar.gz" 
ref_data_sha="9340fj034nf037ngf03g"
ref_pgp_author="???"
validation_method=by_value (by_sha, by_script ...)

[test2.xml]
ref_data_url="gist/tclb/test_scmp_v11.d.tar.gz" 
ref_data_sha="9340fj034nf037ngf03g"
ref_pgp_author="???"
validation_method=run_only

```
some sort of security is _needed_, maybe we could use signed gz? Each of us would need to publish pgp public key. We should also swich to signed commits (but that other part of the story)

**Comment by llaniewski** [comment]
For now the test configuration files look like this:
https://github.com/CFD-GO/TCLB_tests/blob/llaniewski/d2q9/karman.test

@mdzik Do you think we need more security then github repo provides?

**Comment by mdzik** [comment]
For now we use only in-repo files, and rather small ones. If we would like to go for larger sets of reference data (whole vti file?) than some external resources might be needed

```
need karman.xml
need htts://somewhare.com/mdzik-tclb/testsdata/karman.vti 
sha1sum karman.vti sdfasd76fa9s67fa9s86f9as867f
sha1sum karman.xml sdfasd76fa9s67fa9s86f9as867f

run $SOLVER karman.xml
csvdiff output/karman_Log_P00_00000000.csv
diff output/karman_VTK_P00_00001000.pvti
sha1  output/karman_VTK_P00_00001000.vti
```
something along the lines of package build scripts:
https://git.archlinux.org/pacman.git/plain/proto/PKGBUILD.proto

those "scripts" should be run-able locally and we do not want unauthorized code injection (this also goes with main TCLB code).

as for github repo security - it's provided if all commits are signed and if all core-team have 2fa
